packages/dx-scheduler-core/src/plugins/editing-state/reducers.js

```diff
@@ -0,0 +1,23 @@
+export const addAppointment = (
+  addedAppointment, { appointment } = { appointment: {} },
+) => appointment;
+
+export const changeAddedAppointment = (
+  addedAppointment, { change },
+) => ({ ...addedAppointment, ...change });
+
+export const cancelAddedAppointment = () => ({});
+
+export const deleteAppointment = (deletedAppointmentId, { appointmentId }) => appointmentId;
+
+export const cancelDeletedAppointment = () => null;
+
+export const startEditAppointment = (prevEditingAppointmentId, { appointmentId }) => appointmentId;
+
+export const stopEditAppointment = () => null;
+
+export const changeAppointment = (
+  addedAppointment, { change },
+) => ({ ...addedAppointment, ...change });
+
+export const cancelChanges = () => ({});
```

Comment:
Why do we sometimes return an empty object and sometimes null?

Reply:
Because sometimes default value for an object's property is `{}` and sometimes default value for a number's property is `null`.